docs/design/fabric-v2+/fpc-key-dist.puml

```diff
@@ -25,184 +24,153 @@ participant "TLCC_Enclave Peer1_Org1" as TLCC_Enclave1 #99FF99
+      on the creation and registration of the enclave credentials.
+    - Check out the fpc-components diagram for more details
+      on messages, parameters and cryptographic keys used below.
+end note
+
+group chaincode key generation
 
   note right Admin1
     One admin must begin the key
     generation using the local enclave.
     This might as well be Admin2
     using Enclave2.
   end note
-  Admin1        -> Admin1       :   cc_target_message <- <CC_name, CC_ver, CH_id, org_id>
-  Admin1        -> ECC_Manager1 ++: GenerateCCKeys(cc_target_message)
-  ECC_Manager1  -> Storage1     ++: get enclave's sealed_state
-  note right
-    If the enclave is up and running,
-    this step and the sealed_state
-    paramenters below can be skipped.
+  Admin1        -> Admin1       :   fill CC_Parameters structure
+  Admin1        -> E_Manager1   ++: GenerateCCKeys(CC_Parameters)
+
+  E_Manager1    -> ERCC1        ++: check Enclave1 registration
+  return boolean b
+  E_Manager1    -> ECC1         ++: if(b) then GenerateCCKeys(CC_Parameters)
+  ECC1          -> Enclave1     ++: GenerateCCKeys(CC_Parameters)
+  Enclave1      -> Enclave1     :   check input CC_Parameters against internal CC_Parameters
+
+  note over Enclave1
+    It is not necessary to enforce that Enclave1 has already been registered in ERCC.
+    As this is the "first" time that keys are generated,
+    there is no security-critical state and/or other keys to protect.
   end note
-  return /'Storage1 -> ECC_Manager1'/
-
-  ECC_Manager1  -> ECC1         ++: GenerateCCKeys(cc_target_message, sealed_state)
-  ECC1          -> Enclave1     ++: GenerateCCKeys(cc_target_message, sealed_state)
-  Enclave1      -> Enclave1     :   ecc_config <- Unseal(sealed_state)
-  Enclave1      -> Enclave1     :   check input parameters against ecc_config 
-  Enclave1      -> TLCC_Enclave1++: can_endorse(ECC1_S_PK)
-    note left of Enclave1
-        This can_endorse may be unnecessary,
-        because the enclave is registered,
-        and ERCC will check if it can endorse.
-    end note
-    note left of Enclave1 #ff0000
-        The can_endorse function is a nice-to-have feature.
-        We must check how it can be implemented.
-    end note
 
-  return true
-    note left
-        if false, abort
-    end note
-
-  Enclave1      -> Enclave1     :   SEK, CC_E_{PK,SK} <- keyGen
-    note left
-        - SEK denotes the state encryption key
-        - CC_E_{PK,SK} denotes chain code pub/priv key pair
-    end note
-  Enclave1      -> Enclave1     :   add SEK, CC_E_{PK,SK} to ecc_config
-  Enclave1      -> Enclave1     : SIG_ECC1 <- sign <CC_E_PK> with ECC1_S_SK
-    note left
-        Confirmation statement that the enclave has CC_E_SK.
-    end note
-
-  Enclave1      -> Enclave1     :   sealed_state <- Seal(ecc_config)
-  Enclave1      -> Enclave1     :   key_confirmation_message <- <ECC1_S_PK, SIG_ECC1, CC_E_PK>
+  Enclave1      -> Enclave1     :   SEK, Chaincode_DK <- keyGen()
```

Comment:
> we should add a comment on what SEK and Chaincode_DK stand for. I also think DK is a bit misleading as most people think it would be the private key but from further down it is actually the key-pair? Why explicitly list the pair here and the we do not need the extract below.

updated.

> Somewhat alternatively (and probably better) is to do what we have in registration on having a structure in components which lists the individual keys and just add a key-gen which returns a single instance of the structure)

I agree, but this is more meaningful as we detail the sealing of the keys.